rs/registry/admin/bin/main.rs

```diff
@@ -6449,21 +6659,24 @@ impl NnsCanisterClient {
         let mut ids_to_try = vec![self.handler_id];
         ids_to_try.extend(ic_nns_constants::ALL_NNS_CANISTER_IDS.iter().cloned());
 
+        let method = msg.to_string();
+
         for canister_id in ids_to_try {
+            // Convert CanisterId to candid::Principal
+            let principal = candid::Principal::try_from(canister_id.get().as_slice())
+                .map_err(|e| format!("Failed to convert CanisterId to Principal: {:?}", e))?;
+
             let result = self
                 .agent
-                .execute_update(
-                    &canister_id,
-                    &canister_id,
-                    msg.to_string(),
-                    arguments.clone(),
-                    generate_nonce(),
-                )
+                .update(&principal, &method)
+                .with_arg(arguments.clone())
+                .call_and_wait()
                 .await;
 
             match result {
-                Ok(result) => return Ok(result),
-                Err(error_string) => {
+                Ok(response) => return Ok(Some(response)),
```

Comment:
IIUC, it is not possible for this method to return `Ok(None)`. Ideally, in some future PR (maybe never), we would change the return type to just `Result<Vec<u8>, ...>`...